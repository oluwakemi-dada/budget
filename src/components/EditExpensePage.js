import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense, removeExpense } from '../actions/expenses';

const EditExpensePage = ({ history, match }) => {
  const dispatch = useDispatch();

  // SELECTORS
  const expense = useSelector((state) =>
    state.expenses.find((expense) => expense.id === match.params.id)
  );

  // HANDLERS
  const onSubmitHandler = (updatedExpense) => {
    dispatch(editExpense(expense.id, updatedExpense));
    history.push('/');
  };

  const onRemoveHandler = () => {
    dispatch(removeExpense(expense.id));
    history.push('/');
  };

  return (
    <div>
      <ExpenseForm expense={expense} onSubmit={onSubmitHandler} />
      <button onClick={onRemoveHandler}>Remove</button>
    </div>
  );
};

export default EditExpensePage;
