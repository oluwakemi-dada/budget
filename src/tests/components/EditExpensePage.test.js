import React from 'react';
import { shallow } from 'enzyme';
import * as reactRedux from 'react-redux';
import EditExpensePage from '../../components/EditExpensePage';

jest.mock('react-redux', () => ({
  useDispatch: jest.fn(),
  useSelector: jest.fn(),
}));

test('should render EditExpensePage', () => {
  const history = { push: jest.fn() };
  const match = { params: { id: 1 } };
  const wrapper = shallow(<EditExpensePage history={history} match={match} />);
  expect(wrapper).toMatchSnapshot();
});
