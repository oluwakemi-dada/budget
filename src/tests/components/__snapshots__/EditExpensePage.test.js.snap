// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditExpensePage 1`] = `
<div>
  <ExpenseForm
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
